src/mse.ts

```
@@ -17,50 +17,42 @@ export class MSERep extends EventEmitter implements MSE {
 	private pep: PepTalkClient & PepTalkJS
 	private connection?: Promise<PepResponse> = undefined
 
-	private isAwaitingConnection = false
 	private reconnectTimeout?: NodeJS.Timeout = undefined
+	private lastReconnectTime = Date.now()
 
 	constructor(hostname: string, restPort?: number, wsPort?: number, resthost?: string) {
 		super()
 		this.hostname = hostname
 		this.restPort = typeof restPort === 'number' && restPort > 0 ? restPort : 8580
 		this.wsPort = typeof wsPort === 'number' && wsPort > 0 ? wsPort : 8595
 		this.resthost = resthost // For ngrok testing only
-		this.pep = startPepTalk(this.hostname, this.wsPort)
-		this.pep.on('close', () => this.onPepClose())
-		this.connection = this.pep.connect()
+		this.pep = this.initPep()
+	}
+
+	initPep(): PepTalkClient & PepTalkJS {
+		const pep = startPepTalk(this.hostname, this.wsPort)
+		pep.on('close', () => this.onPepClose())
+		this.connection = pep.connect().catch((e) => e)
```

I'm not particular fond of this pattern. Catching and silently swallowing the error is a bug waiting to happen.

If it where me, I would remove the `createMSE()` function altogether and make it so that the consumer would instead:

```
const myMSE = new MSE(...options)
myMSE.on('connected', handleConnected)
myMSE.on('disconnected', handleDisconnected)
try {
  await myMSE.ini()
} catch (e) {
  handleInitialError(e)
}
// later:
myMSE.dispose() // disconnect and stop reconnecting
```

But that is a breaking change I suppose..

---

I agree this needs a slightly larger rewrite